clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -6725,16 +6722,17 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
   // the command line.  This is to allow for CMake based builds using the
   // Linux based driver on Windows to correctly pull in the expected debug
   // library.
-  if (Args.hasArg(options::OPT_fsycl) && !Args.hasArg(options::OPT_nolibsycl)) {
-    if (!D.IsCLMode() && TC.getTriple().isWindowsMSVCEnvironment()) {
+  if (Args.hasArg(options::OPT_fsycl) && !Args.hasArg(options::OPT_nolibsycl) &&
+      !D.IsCLMode()) {
```

Comment:
What difference does it make by combining just `!D.IsCLMode()` in this `if` statement ?
Why not both `!D.IsCLMode() && TC.getTriple().isWindowsMSVCEnvironment()` ?

Reply:
This is due to the difference in behavior needed for line#6727 and line#6735.